### PR TITLE
Dev/rm/book 8 add endpoint reservation

### PR DESCRIPTION
**Pull request by zRalphy**

### CODE REVIEW

File: src/main/resources/database/2021-10-05/11-create-reservation.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+--changeset rmecwaldowski:4
+
+CREATE TABLE reservation(
+	id BIGINT AUTO_INCREMENT PRIMARY KEY,
+    status ENUM (IN_PROGRESS, REALIZED, CANCELED),
+    flight_id BIGINT NOT NULL
+);
+
+ALTER TABLE reservation
+             ADD CONSTRAINT reservation_id
```
Comment (gstudzinski) [inline src/main/resources/database/2021-10-05/11-create-reservation.sql:11]: this FK constraint can be done within CREATE TABLE
are you using right id for FOREIGN KEY (id)?

File: src/main/resources/database/2021-10-05/12-edit-passenger.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+--liquibase formatted sql
+--changeset rmecwaldowski:4
+
+   ALTER TABLE passenger
+         ADD CONSTRAINT passenger_id
+         FOREIGN KEY (id) REFERENCES reservation(id)
```
Comment (gstudzinski) [inline src/main/resources/database/2021-10-05/12-edit-passenger.sql:6]: same are you using right id for FOREIGN KEY (id)?
and you need additional column in passanger for FK

File: src/main/java/com/pgs/booking/model/Reservation.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    @Id
+    @GeneratedValue
+    @Column(name = "id")
+    private Long id;
+    @Column(name = "status")
+    private ReservationStatus status;
+
+    @ManyToOne(fetch = FetchType.LAZY, optional = false, cascade = CascadeType.PERSIST)
+    @JoinColumn(name = "flight_id", nullable = false)
+    private Long flight_id;
```
Comment (gstudzinski) [inline src/main/java/com/pgs/booking/model/Reservation.java:26]: is this Long type ok? please check in the documentation

File: src/main/java/com/pgs/booking/model/dto/CreateUpdateReservationDto.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+public class CreateUpdateReservationDto {
+
+    private Reservation.ReservationStatus status;
+    private Long flight_id;
+    private List<CreateUpdatePassengerDto> passengers;
```
Comment (gstudzinski) [inline src/main/java/com/pgs/booking/model/dto/CreateUpdateReservationDto.java:18]: if we require passengers (I guess yes)  set proper validation annotation for list

File: src/main/java/com/pgs/booking/model/dto/ReservationDto.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.Value;
+
+import java.util.List;
+
+@Value
+@Builder
+public class ReservationDto {
+    Long id;
+    Reservation.ReservationStatus status;
+    Long flight_id;
```
[the remark this answers is not in the document]
Comment (zRalphy) [reply src/main/java/com/pgs/booking/model/dto/ReservationDto.java:14]: I will change it.

File: src/main/java/com/pgs/booking/mappers/CreateUpdateReservationDtoMapper.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class CreateUpdateReservationDtoMapper {
+
+    private final CreateUpdatePassengerDtoMapper createUpdatePassengerDtoMapper;
+
+    public List<CreateUpdateReservationDto> mapToCreateReservationsDto(List<Reservation> reservations) {
```
Comment (pszerszen) [inline src/main/java/com/pgs/booking/mappers/CreateUpdateReservationDtoMapper.java:17]: not used

File: src/main/java/com/pgs/booking/mappers/CreateUpdateReservationDtoMapper.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        return reservations.stream()
+                .map(this::mapToCreateReservationDto)
+                .collect(Collectors.toList());
+    }
+
+    public List<Reservation> mapToReservations(List<CreateUpdateReservationDto> createUpdateReservationsDto) {
```
Comment (pszerszen) [inline src/main/java/com/pgs/booking/mappers/CreateUpdateReservationDtoMapper.java:23]: not used

File: src/main/java/com/pgs/booking/mappers/ReservationDtoMapper.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class ReservationDtoMapper {
+
+    private final PassengerDtoMapper passengerDtoMapper;
+
+    public List<ReservationDto> mapToPassengersDto(List<Reservation> reservations) {
```
Comment (pszerszen) [inline src/main/java/com/pgs/booking/mappers/ReservationDtoMapper.java:16]: not used

File: src/main/java/com/pgs/booking/model/dto/CreateUpdateReservationDto.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+public class CreateUpdateReservationDto {
+    @NotNull
+    private Reservation.ReservationStatus status;
+    @NotNull
+    private Flight flight;
+    private List<@NotNull CreateUpdatePassengerDto> passengers;
```
Comment (pszerszen) [inline src/main/java/com/pgs/booking/model/dto/CreateUpdateReservationDto.java:20]: I think you should also validate if the list is not empty

File: src/main/java/com/pgs/booking/model/Reservation.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+@Builder
+@Entity
+@Table(name = "reservation")
+public class Reservation implements Serializable {
+    @Id
+    @GeneratedValue
+    @Column(name = "id")
+    private Long id;
+    @Column(name = "status")
+    private ReservationStatus status;
```
Comment (pszerszen) [inline src/main/java/com/pgs/booking/model/Reservation.java:23]: doesn't it miss `@Enumerated(EnumType.STRING)`?

File: pom.xml [hunk 11]
```diff
@@ -78,16 +78,22 @@
...
 
+        <dependency>
             <groupId>org.springframework.security</groupId>
             <artifactId>spring-security-test</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-core</artifactId>
+            <version>5.2.2.Final</version>
```
Comment (pszerszen) [inline pom.xml:95]: specifying the version here is redundant here as spring-boot does it for us

File: src/main/java/com/pgs/booking/service/ReservationService.java [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+    public ReservationDto getReservation(long id) {
+        return null;
+    }
+
+    public ReservationDto addReservation(CreateUpdateReservationDto createUpdateReservationDto) {
+
+        var flight_id = createUpdateReservationDto.getFlight();
+        var passengersDtoList = passengerService.getPassengers();
+        //var passengersList = createUpdatePassengerDtoMapper.mapToPassengers(passengersDtoList);
```
Comment (pszerszen) [inline src/main/java/com/pgs/booking/service/ReservationService.java:44]: do not leave commented code

File: src/main/java/com/pgs/booking/service/ReservationService.java [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+    private final PassengerService passengerService;
+    private final CreateUpdatePassengerDtoMapper createUpdatePassengerDtoMapper;
+    private final PassengerDtoMapper passengerDtoMapper;
+
+    public List<PassengerDto> getPassengersByReservation(long id) {
+        return passengerDtoMapper.mapToPassengersDto(reservationRepository.getPassengersByReservation(id));
+    }
+
+    public ReservationDto getReservation(long id) {
+        return null;
```
Comment (pszerszen) [inline src/main/java/com/pgs/booking/service/ReservationService.java:37]: implement

File: src/main/java/com/pgs/booking/service/ReservationService.java [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public Reservation.ReservationStatus setReservationStatusCanceled(Reservation.ReservationStatus reservationStatus) {
+        return reservationStatus;
+    }
+
+    public Reservation.ReservationStatus setReservationStatusRealized(Reservation.ReservationStatus reservationStatus) {
+        return reservationStatus;
+    }
+
```
Comment (pszerszen) [inline src/main/java/com/pgs/booking/service/ReservationService.java:63]: What is a purpose of those methods? Their name suggests setting a specific status, while they accept a status as a parameter... Adn where (on what object) are they supposed to set the status?

File: src/main/java/com/pgs/booking/service/ReservationService.java [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+import com.pgs.booking.repository.PassengerRepository;
+import com.pgs.booking.repository.ReservationRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class ReservationService {
```
Comment (pszerszen) [inline src/main/java/com/pgs/booking/service/ReservationService.java:21]: missing unit test

File: src/main/java/com/pgs/booking/model/dto/CreateUpdateReservationDto.java (hunk 4 above)
Comment (zRalphy) [reply src/main/java/com/pgs/booking/model/dto/CreateUpdateReservationDto.java]: changed and outdated.

File: src/main/java/com/pgs/booking/model/dto/CreateUpdateReservationDto.java [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+public class CreateUpdateReservationDto {
+
+    private Reservation.ReservationStatus status;
```
[the remark this answers is not in the document]
Comment (zRalphy) [reply src/main/java/com/pgs/booking/model/dto/CreateUpdateReservationDto.java]: changed and outdated.

File: src/main/java/com/pgs/booking/model/dto/ReservationDto.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (zRalphy) [reply src/main/java/com/pgs/booking/model/dto/ReservationDto.java]: changed and outdated.

**Pull request merged**